module/zfs/zio.c

```diff
@@ -2112,7 +2112,7 @@ zio_suspend(spa_t *spa, zio_t *zio)
 		    ZIO_FLAG_CANFAIL | ZIO_FLAG_SPECULATIVE |
 		    ZIO_FLAG_GODFATHER);
 
-	spa->spa_suspended = B_TRUE;
+	spa->spa_suspended = reason;
```

Comment:
The `spa->spa_suspended` type should be changes from a `uint8_t` to a `zio_suspend_reason_t`.

Reply:
Fixed